app/scripts/controllers/transactions/index.js

```diff
@@ -1293,7 +1305,11 @@ export default class TransactionController extends EventEmitter {
     // we need to keep track of what is currently being signed,
     // So that we do not increment nonce + resubmit something
     // that is already being incremented & signed.
-    if (this.inProcessOfSigning.has(txId)) {
+    const txMeta = this.txStateManager.getTransaction(txId);
+    if (
+      this.inProcessOfSigning.has(txId) ||
+      txMeta.status === TRANSACTION_STATUSES.SUBMITTED
+    ) {
```

Comment:
The check is added to ensure that the fn does not re-approve a transaction `txMeta.status === TRANSACTION_STATUSES.SUBMITTED`.